### DISCUSSION TITLE
AIO + Memories - add perl to nextcloud container

### DISCUSSION THREAD

**Discussion by frederikb96** [nextcloud/all-in-one#3449] [Ideas] [answered]
Currently, the Nextcloud container does not have perl installed, which is necessary for Memories exiftool to work. The container already contains ffmpeg which is also necessary for Memories. To enable easier installation of Memories within Nextcloud AIO, could make sense to add perl to the Docker Image :)

**Comment by szaimen** [comment]
Hi @pulsejet sorry for pinging you but I thought Perl was bundled in Memories or did I misunderstand something?

**Reply by pulsejet** [reply]
Thanks for the ping @szaimen. Memories does include Perl for x86 and arm64. The only place where Perl would be required is likely BSD or on 32-bit hardware. Are you running into some issue with the bundled binaries @frederikb96?

**Reply by szaimen** [reply]
Thanks for the answer @pulsejet :)
AIO does not run on BSD nor 32-bit so that should not be the case here?

**Reply by szaimen** [reply]
What about Alpine? Could that be a problem?

**Reply by pulsejet** [reply]
Alpine should be fine too; both musl and glibc binaries are bundled in.

**Reply by szaimen** [reply]
okay then this is weird

**Reply by frederikb96** [reply, marked as answer]
Hm, I removed Perl from my additional packages for AIO and then the admin interface of Memories said that exiftool is not usable since it depends on Perl. So I added Perl again to AIO.

Maybe some issue because it was first available and then removed by me. So probably not a problem for fresh installations :)

**Reply by s8m** [reply]
Perl is missing from fresh install of AIO v7.9.0 (NC v28.0.1):

![image](https://github.com/nextcloud/all-in-one/assets/5231279/94ee31d7-b486-4d55-998c-4395c6705f89)

**Reply by pulsejet** [reply]
What's the error if you turn off `use system perl`?

**Reply by s8m** [reply]
The red alert disappears, but the "perl binary not found" message persists:

![image](https://github.com/nextcloud/all-in-one/assets/5231279/8dd4fcb2-4337-49d7-bf03-9cb73bf5f539)

**Reply by pulsejet** [reply]
As the **warning** says, you need perl only if the packaged binary doesn't work.

So ignore it.
